cellxgene_schema_cli/cellxgene_schema/validate.py

```diff
@@ -1501,6 +1502,29 @@ def _validate_spatial_is_primary_data(self):
                 "When uns['spatial']['is_single'] is False, obs['is_primary_data'] must be False for all rows."
             )
 
+    def _validate_spatial_assay_ontology_term_id(self):
+        """
+        If assay is spatial, all assay ontology term ids should be identical.
+
+        :rtype none
+        """
+        # Identical check requires assay_ontology_term_id.
+        obs = getattr_anndata(self.adata, "obs")
+        if obs is None or "assay_ontology_term_id" not in obs:
+            return
+
+        # Identical check is only applicable to spatial datasets.
+        if not self._is_supported_spatial_assay():
+            return
+
+        # Validate assay ontology term ids are identical.
+        term_count = obs["assay_ontology_term_id"].nunique()
+        if term_count > 1:
+            self.errors.append(
+                "When obs['assay_ontology_term_id'] is either 'EFO:0010961' (Visium Spatial Gene Expression) or "
```

Comment:
If I add 'EFO:0009922' (10x 3' v3) as an `assay_ontology_term_id` for one obs in a spatial dataset, the validation fails but I don't get this error or any errors